src/distilabel/steps/base.py

```diff
@@ -43,6 +46,37 @@
 extra metadata that allows `distilabel` to perform validations over the `process` step
 method defined in each `Step`"""
 
+# Pattern to convert PascalCase to snake_case
+PATTERN_PASCAL_NAME = re.compile(r"(?<!^)(?=[A-Z])")
+
+
+def _infer_step_name(step_cls_name: str, pipeline: Optional["Pipeline"]) -> str:
+    """Infer the name of the step based on the class name and the pipeline.
+
+    If a `Pipeline` is given (the general case), it will check if the name already exists
+    in the steps of the `DAG`, to add a number at the end of the name.
+
+    Args:
+        step_cls_name: The step class name, as obtained by `type(cls).__name__`.
+        pipeline: The `Pipeline` the step belongs to, can be `None` if the step is created
+            outside of a `Pipeline`.
+
+    Returns:
+        A name for the step.
+
+    Example:
+        ```python
+        >>> _infer_step_name("StepWithOnePreviousStep", None)
+        'step_with_one_previous_step'
+        ```
+    """
+    name = re.sub(PATTERN_PASCAL_NAME, "_", step_cls_name).lower() + "_0"
+    if pipeline:
+        # Check the name doesn't already exist in the pipeline
+        if name in set(pipeline.dag.G):
+            name = f"{name[:-1]}{int(name[-1])+1}"
```

Comment:
we need a while condition here, it can happen that after adding 1, there is another step called like that too. Also, adding 1 to `int(name[-1])` works from 0 to 9, but not for numbers with more than 1 digit.

Reply:
totally